src/components/SearchTab.tsx

```diff
@@ -0,0 +1,89 @@
+import { IAlbum, ISong } from 'dxmp-common';
+import * as React from 'react';
+
+import { SongResult } from './SongResult';
+
+import './SearchTab.scss';
+
+interface ISearchTabProps {
+  albums: {[albumID: number]: IAlbum},
+  songs: ISong[],
+};
+
+interface ISearchTabState {
+  query: string,
+  queryResults: ISong[],
+  scrollTop: number,
+};
+
+const RESULT_HEIGHT = 80;
+const RESULTS_PER_ROW = 2;
+const TOP_BUFFER = 1;
+const BOTTOM_BUFFER = 7;
+
+export class SearchTab extends React.Component<ISearchTabProps, ISearchTabState> {
+  public state: ISearchTabState = {
```

Comment:
Had no idea you could do default assignment like this.

![](https://media.giphy.com/media/l0HlJ0wSmbfCZTd96/giphy.gif)